ts/Core/Foundation.ts

```diff
@@ -0,0 +1,87 @@
+/* *
+ *
+ *  (c) 2010-2021 Torstein Honsi
+ *
+ *  License: www.highcharts.com/license
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ * */
+
+'use strict';
+
+/* *
+ *
+ *  Imports
+ *
+ * */
+import Axis from 'Axis/Axis.js';
+import Series from 'Series/Series.js';
+
+import U from './Utilities.js';
+const {
+    addEvent,
+    isFunction,
+    objectEach,
+    removeEvent
+} = U;
+
+
+/* *
+ *
+ *  Functions
+ *
+ * */
+
+/*
+ * Register event options. If an event handler is set on the options, it should
+ * be subject to Axis.update and Series.update. This is contrary to general
+ * handlers that are set directly using addEvent either on the class or on the
+ * instance. #6943, #10861.
+ */
+const registerEventOptions = (
+    component: Axis|Series
+): void => {
+
+    // A lookup over those events that are added by _options_ (not
+    // programmatically). These are updated through .update()
+    component.eventOptions = component.eventOptions || {};
+
```

Comment:
I was expecting a class structure.
```ts
class Foundation {
  eventOptions: Record<string, Function>;
  protected registerEventOptions(eventOptions: Record<string, Function>): void {…};
}
export default Foundation;
```
```ts
class Axis extends Foundation {
…
    this.registerEventOptions(this.options.events);
…
}
```
```ts
class Series extends Foundation {
…
    this.registerEventOptions(this.options.events);
…
}
```

Reply:
Yes, eventually, when we see more patterns... Isn't this okay for a start, since we only have one function to share between the two classes?

Reply:
It is for me. I just fear, that someone else will come along and add to the temporary pattern, and eventually it will become difficult to refactor it as a class.

Reply:
I suggest to wrap it in a static class function.
```ts
class Foundation {
  public static registerEventOptions = (
    component: Axis|Series
  ): void => {…}
}
export default Foundation;
```